### ISSUE TITLE
Change bugs page on browser to github issues.

### ISSUE THREAD

**Issue by chimosky** [sugarlabs/browse-activity#117]
We moved from the bug tracker to using github issues so I think the bugs hyperlink in the activity home page should be changed to our github issues.

**Comment by quozl** [comment]
Not everyone moved.  trac is still used by some contributors.  Once trac required an account, the bug reports fell quite a bit.  A new way to gain feedback is needed.  I don't feel GitHub issues is meeting this need, but if it is all we want to offer, then that's what should be linked.

**Comment by chimosky** [comment]
I agree, github issues isn't all we want to offer since we still have trac and it'll make sense to include it in the home page as not a lot of people use trac these days and we wouldn't need people opening new issues there as it'll be easier for those issues to be fixed if they're opened in github.

**Comment by quozl** [comment]
Easier if opened on GitHub?  Why is that?  I find problems are just as easy to fix if they are opened on trac or on GitHub issues.

**Comment by chimosky** [comment]
A lot of our contributors would most likely find it easier to use Github and would most likely come to Github first to find issues, so by easier to fix I mean easier to find - should've used the proper wording there -.

**Comment by quozl** [comment]
Yes, but now we ought to optimise the reporting of problems rather than focus on a relatively minor aspect of the work of the contributors.

Almost all our issues are being created by code contributors; people who could very well have created a pull request instead.

With this in mind, I've reviewed the browser start page data/index.html;

* there's no call to action, e.g. "report a problem",

* there's no translations to the most frequently used non-english languages,

* there's no warning that a GitHub account will be required, and therefore a mail account, (presumably it is still possible to create GitHub accounts without 2FA),

Also on review, I'm puzzled as to why the links and the other content are written into the document by a Javascript fragment; but that could be a great way to handle the languages.

**Comment by chimosky** [comment]
> Yes, but now we ought to optimize the reporting of problems rather than focus on a relatively minor aspect of the work of the contributors.

Yeah I agree about optimizing reporting of problems but I also think that the relatively minor aspect of the work of the contributors can sometimes be made easier as sometimes it takes people a while to find something or to understand something. 

> Almost all our issues are being created by code contributors; people who could very well have created a pull request instead.

I think that it takes some contributors some time to get the hang of things and the also considering that some are just encountering the code base for the first time but sending a PR does depend on their level of expertise and that varies.


> With this in mind, I've reviewed the browser start page data/index.html;
>
> * there's no call to action, e.g. "report a problem",

Agreed.

> * there's no translations to the most frequently used non-english languages,

I see translations for French and Spanish.

> * there's no warning that a GitHub account will be required, and therefore a mail account, (presumably it is still possible to create GitHub accounts without 2FA),

Yeah I think a Github account can be created without 2FA, it might not be necessary to include a warning for a Github account being needed as when you visit the Github page, the sign up and sign in buttons are clear at the top of the page.

> Also on review, I'm puzzled as to why the links and the other content are written into the document by a Javascript fragment; but that could be a great way to handle the languages.

Have little JavaScript knowledge but I think it's easier as they're just hyperlinks.

**Comment by quozl** [comment]
> I see translations for French and Spanish.

I can't.  Look again?  See file data/index.html

**Comment by chimosky** [comment]
> > I see translations for French and Spanish.
> 
> I can't. Look again? See file data/index.html

Ah! I was talking about the po files.

**Comment by quozl** [comment]
So we have, from good to perfect;

* [ ] change bugs.sugarlabs.org to github.com/sugar/ issues,
* [ ] change the link text to "report a problem",
* [ ] provide translations of the link text,
* [ ] deploy a software project feedback system, such as [LogChimp](https://github.com/logchimp/logchimp) or [Astuto](https://github.com/riggraz/astuto), along with human contributors to review the feedback,
* [ ] create a feedback activity for capturing data while the end user is not connected,
* [ ] consult with Music Blocks and Sugarizer in case their feedback systems might be repurposed.

For your interest, OLPC OS contains a telemetry client that transmits activity start and stop data to OLPC servers.  For Fedora 18, it is part of the activity.  For Ubuntu 20.04, it is part of Sugar Toolkit.

**Comment by chimosky** [comment]
> For your interest, OLPC OS contains a telemetry client that transmits activity start and stop data to OLPC servers. For Fedora 18, it is part of the activity. For Ubuntu 20.04, it is part of Sugar Toolkit.

Is the code for the client just included in your images or it's also upstream?

**Comment by quozl** [comment]
We can *not want to use GitHub issues as the only means of reporting bugs*, yet at the same time only offer GitHub issues on the Browse default page.  I imagine a proper feedback system will take a while.  Don't let the perfect be the enemy of the good.  I've sent you the telemetry code by private mail.

**Comment by chimosky** [comment]
> We can _not want to use GitHub issues as the only means of reporting bugs_, yet at the same time only offer GitHub issues on the Browse default page. I imagine a proper feedback system will take a while. Don't let the perfect be the enemy of the good. I've sent you the telemetry code by private mail.

At the moment trac is what's offered as a means of reporting bugs on the Browse default page, the aim is to include GitHub issues. I agree that a proper feedback system will take a while.

**Comment by Dimi20cen** [comment]
So for starters, we should change the link text from "bugs" to "report a problem" which should link to "github.com/sugar/ issues", is that right?

**Comment by quozl** [comment]
Thanks for reading the discussion.  Since my earlier comments, Sugar Labs has shrunk and we don't have people with the time to handle bug reports.  This is an organisational governance issue.  I think "bugs" should be removed and not replaced with anything, so I've just made that change.  Users are unlikely to go into Browse to discover how to give feedback, but there are no feedback options in Sugar _apart from_ Browse.

* 6ff2e9a ("Remove bugs.sugarlabs.org")

Next actions;
* [ ] organisation to determine how to solicit and handle feedback.

**Comment by Dimi20cen** [comment]
I was thinking that maybe we could add a "help"/"report a bug" option on the menu that opens up when you press the xo icon,, that could link to an overall issue page on github. But from what you said, I'm understanding that the issue of soliciting and handling feedback isn't  a priority at the moment, right?

**Comment by quozl** [comment]
It could also be in the _My Settings_ as a panel.  But still, how would you authenticate the child or teacher?  In the environment we face now, any feedback mechanism without authentication is used to send unsolicited commercial mail.  The Discourse 1.8.8 instance (socialhelp.sugarlabs.org in 2014 to 2017) was an attempt to handle this, but did not catch on, because it did not have wide acceptance of the members of the organisation.  That's another reason why we should step back from this and let the organisation make the decision.  GitHub is what we use now, but we may switch to GitLab if the terms and conditions of GitHub become unwelcoming.

**Comment by Dimi20cen** [comment]
Hm ok I hadnt even thougth of mail spam.

**Comment by arav-behl** [comment]
Hey, I am new to Open Source but I want to contribute to SugarLabs as I am passionate about Education, and I'm also doing research on how education can be improved using AI in my university. Thus, I feel SugarLabs is a great project and I hope I can be assigned some beginner friendly issues on which I can contribute :)

**Comment by chimosky** [comment]
@arav-behl you might want to take a look at our [docs](https://github.com/sugarlabs/sugar-docs).
